srcpkgs/luarocks/template

```diff
@@ -1,38 +1,77 @@
 # Template file for 'luarocks'
 pkgname=luarocks
-version=2.4.4
+version=3.0.4
 revision=1
 archs=noarch
 build_style=configure
-configure_args="
- --prefix=/usr
- --sysconfdir=/etc/luarocks
- --lua-version=5.3
- --lua-suffix=5.3
- --with-lua-include=/usr/include/lua5.3
- --with-downloader=curl
+configure_args=" --prefix=/usr --sysconfdir=/etc
+ --lua-version=5.3 --with-lua-include=/usr/include/lua5.3
  --versioned-rocks-dir"
 hostmakedepends="lua-devel"
 depends="lua curl unzip"
-conf_files="/etc/luarocks/config-5.3.lua"
 short_desc="Package management for Lua modules"
 maintainer="Duncaen <duncaen@voidlinux.org>"
 license="MIT"
-homepage="http://luarocks.org/"
-distfiles="http://luarocks.org/releases/luarocks-${version}.tar.gz"
-checksum="3938df33de33752ff2c526e604410af3dceb4b7ff06a770bc4a240de80a1f934"
+homepage="https://luarocks.org/"
+distfiles="https://luarocks.org/releases/luarocks-${version}.tar.gz"
+checksum=1236a307ca5c556c4fed9fdbd35a7e0e80ccf063024becc8c3bf212f37ff0edf
 
-alternatives="
- luarocks:luarocks:/usr/bin/luarocks-5.3
- luarocks:luarocks-admin:/usr/bin/luarocks-admin-5.3"
+conf_files="/etc/luarocks/config-5.3.lua"
 
 pre_configure() {
-	sed -i -e '/unzip/d' configure
+	vsed -i -e '457,465d' configure
 }
+
+post_build() {
+	make LUA_VERSION=5.2 LUA_INTERPRETER=lua5.2 \
+		 LUA_INCDIR=/usr/include/lua5.2 ./build/config-5.2.lua
+
+	make LUA_VERSION=5.1 LUA_INTERPRETER=lua5.1 \
+		 LUA_INCDIR=/usr/include/lua5.1 ./build/config-5.1.lua
+}
+
 pre_install() {
-	sed -i -e 's;env lua;env lua5.3;g' src/bin/luarocks{,-admin}
+	vsed -i -e 's;env lua;env lua5.3;g' src/bin/luarocks{,-admin}
 }
+
 post_install() {
 	vlicense COPYING
-	rm -rfv ${DESTDIR}/usr/bin/luarocks{,-admin}
+
+	make DESTDIR="${DESTDIR}" LUA_VERSION=5.1 install-config
+	make DESTDIR="${DESTDIR}" LUA_VERSION=5.2 install-config
```

Comment:
I would move this into the for loop, and the other make invocation too.

Reply:
done